Sources/Views/MusicProgressSliderCell.m

```diff
@@ -0,0 +1,55 @@
+//
+//  MusicProgressSlider.m
+//  Hermes
+//
+//  Created by Xinhong LIU on 19/4/15.
+//
+//
+
+#import "MusicProgressSliderCell.h"
+
+@implementation MusicProgressSliderCell
+
+@synthesize knobImage = _knobImage;
+
+- (NSImage *)knobImage {
+  return _knobImage;
+}
+
+- (void)setKnobImage:(NSImage *)knobImage {
+  _knobImage = knobImage;
+  if (_knobImage == nil) {
+    // default knob image
+    _knobImage = [NSImage imageNamed:@"knob.png"];
+  }
+}
+
+- (id)initWithCoder:(NSCoder *)aDecoder {
+  self = [super initWithCoder:aDecoder];
+  if (self) {
+    [self setKnobImage:[NSImage imageNamed:@"knob.png"]];
+  }
+  return self;
+}
+
+- (NSRect)knobRectFlipped:(BOOL)flipped {
+  float KNOB_WIDTH = [[self knobImage] size].width;
+  float KNOB_HEIGHT = [[self knobImage] size].height;
+  CGFloat percentage = ([self doubleValue]  - [self minValue])
+  / ([self maxValue] - [self minValue]);
```

Comment:
This should fit okay on one line and is easier to read using the dot notation:

`CGFloat percentage = (self.doubleValue - self.minValue) / (self.maxValue - self.minValue);`

In general in most (or all) C languages, a hanging indent is preferred, so if you really want to keep this on two lines I suggest writing it like this:

```
CGFloat percentage = ([self doubleValue] - [self minValue])
    / ([self maxValue] - [self minValue]);
```

but still, I think one line looks great, with the dot notation.